src/crypto/clu_decrypt.c

```diff
@@ -0,0 +1,223 @@
+/* clu_decrypt.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <wolfclu/clu_header_main.h>
+#include <wolfclu/clu_log.h>
+#include <wolfclu/clu_optargs.h>
+#include <wolfclu/genkey/clu_genkey.h>
+
+#define MAX_LEN             1024
+
+/* Used for algorithms that do not have an EVP type
+ * return WOLFCLU_SUCCESS on success
+ */
+int wolfCLU_decrypt(int alg, char* mode, byte* pwdKey, byte* key, int size,
+        char* in, char* out, byte* iv, int block, int keyType)
+{
+#ifdef HAVE_CAMELLIA
+    Camellia camellia;                  /* camellia declaration */
+#endif
+
+    XFILE  inFile;                      /* input file */
+    XFILE  outFile;                     /* output file */
+
+    WC_RNG     rng;                     /* random number generator */
+    byte*   input  = NULL;              /* input buffer */
+    byte*   output = NULL;              /* output buffer */
+    byte    salt[SALT_SIZE] = {0};      /* salt variable */
+
+    int     currLoopFlag = 1;           /* flag to track the loop */
+    int     lastLoopFlag = 0;           /* flag for last loop */
+    int     ret          = 0;           /* return variable */
+    int     keyVerify    = 0;           /* verify the key is set */
+    int     i            = 0;           /* loop variable */
+    int     pad          = 0;           /* the length to pad */
+    int     length;                     /* length of message */
+    int     tempMax = MAX_LEN;              /* equal to MAX_LEN until feof */
+    int     sbSize = SALT_SIZE + block; /* size of salt and iv together */
```

Comment:
I don't love the `sbSize` name. Doesn't really tell me what it is. Suggest `saltAndIvSize`.